osu.Game/Screens/Play/HUD/GameplayLeaderboard.cs

```diff
@@ -46,42 +75,114 @@ protected override void LoadComplete()
         /// Whether the player should be tracked on the leaderboard.
         /// Set to <c>true</c> for the local player or a player whose replay is currently being played.
         /// </param>
-        public ILeaderboardScore AddPlayer([CanBeNull] User user, bool isTracked)
+        public ILeaderboardScore Add([CanBeNull] User user, bool isTracked)
         {
             var drawable = CreateLeaderboardScoreDrawable(user, isTracked);
 
+            if (isTracked)
+            {
+                if (trackedScore != null)
+                    throw new InvalidOperationException("Cannot track more than one score.");
+
+                trackedScore = drawable;
+            }
+
             drawable.Expanded.BindTo(Expanded);
 
-            base.Add(drawable);
+            Flow.Add(drawable);
             drawable.TotalScore.BindValueChanged(_ => sorting.Invalidate(), true);
 
-            Height = Count * (GameplayLeaderboardScore.PANEL_HEIGHT + Spacing.Y);
+            int displayCount = Math.Min(Flow.Count, maxPanels);
+            Height = displayCount * (GameplayLeaderboardScore.PANEL_HEIGHT + Flow.Spacing.Y);
+            requiresScroll = displayCount != Flow.Count;
 
             return drawable;
         }
 
+        public void Clear()
+        {
+            Flow.Clear();
+            trackedScore = null;
+            scroll.ScrollToStart(false);
+        }
+
         protected virtual GameplayLeaderboardScore CreateLeaderboardScoreDrawable(User user, bool isTracked) =>
             new GameplayLeaderboardScore(user, isTracked);
 
-        public sealed override void Add(GameplayLeaderboardScore drawable)
+        protected override void Update()
         {
-            throw new NotSupportedException($"Use {nameof(AddPlayer)} instead.");
+            base.Update();
+
+            if (requiresScroll && trackedScore != null)
+            {
+                float scrollTarget = scroll.GetChildPosInContent(trackedScore) + trackedScore.DrawHeight / 2 - scroll.DrawHeight / 2;
+                scroll.ScrollTo(scrollTarget, false);
```

Comment:
```suggestion
                float scrollTarget = scroll.GetChildPosInContent(trackedScore) + trackedScore.DrawHeight / 2 - scroll.DrawHeight / 2;
                scroll.ScrollTo(Math.Clamp(scrollTarget, 0, scroll.ScrollableExtent), false);
```

Changes the gameplay leaderboard to not scroll beyond the scrollable limits. This allows the effect of the tracked user climbing up to a static top, rather than continuing scrolling the leaderboard beyond 0 making the player at the center even when at number one, and only displaying half of the maximum displayable players.

Also makes the user look more like it's climbing when at the bottom.

before:

https://user-images.githubusercontent.com/22781491/129415990-567f4175-cb43-4102-9557-54fc70eff3c5.mp4


after:

https://user-images.githubusercontent.com/22781491/129415736-8b6043ac-e430-41cb-81b5-c46c592dd129.mp4

Although that makes the scroll to the top not smooth due to being clamped to `0`, I have tried enabling scroll animation but that seems to make the leaderboard a bit too wiggly when scrolling in the middle of the scores.

Reply:
I think the proposal above is what may eventually be desired but it looks bad as provided. So I'm personally leaning towards getting this pull in as is and working on the above as a follow up.

Reply:
Fair, doesn't seem to be too annoying currently so can be left for a follow-up.

Reply:
It's definitely what we want, but let's leave it as a follow up. One would hope it's a bit smoother in transition (some heuristic would need to be added to decide whether to send `animated` or not, based on the target position).